### PR TITLE
Add support for TypeScript 2.7.2 when parsing the output from tsc

### PR DESCRIPTION
**Pull request by boyanio**
This change supports older version of the TypeScript compiler as they have a different output when in watch mode. In version 2.7.2, the compiler always writes `Compilation complete. Watching for file changes.` even though there are errors. So one needs to keep track of the errors as they occur.

### CODE REVIEW

File: lib/stdout-manipulator.js [hunk 1]
```diff
@@ -64,10 +64,17 @@ function manipulate(buffer) {
 
 function detectState(lines) {
   const clearLines = removeColors(lines);
-  const compilationError = clearLines.some(line => compilationCompleteWithErrorRegex.test(line));
+  const newCompilation = clearLines.some(line => newCompilationRegex.test(line));
+  const compilationError = clearLines.some(
```
Comment (gilamran) [inline lib/stdout-manipulator.js]: In the original code the test was done on the `compilationCompleteWithErrorRegex` and it was enough to detect error, why did you add the `typescriptErrorRegex` and `typescriptPrettyErrorRegex` to the test?

File: lib/tsc-watch.js [hunk 2]
```diff
@@ -50,12 +51,15 @@ tscProcess.stdout.on('data', buffer => {
   const lines = manipulate(buffer);
   print(noColors, noClear, lines);
   const state = detectState(lines);
+  const newCompilation = state.newCompilation;
   const compilationError = state.compilationError;
   const compilationComplete = state.compilationComplete;
 
+  compilationErrorSinceStart = (!newCompilation && compilationErrorSinceStart) || compilationError;
```
Comment (gilamran) [inline lib/tsc-watch.js]: Why would you need to test for `newCompilation`? `compilationError` is not enough?
I don't really understand your intents here. please explain

File: lib/stdout-manipulator.js (hunk 1 above)
Comment (boyanio) [reply lib/stdout-manipulator.js]: Because in older versions of TypeScript `compilationCompleteWithErrorRegex ` does not exist. Se my comment below

**Comment by boyanio** [comment]
I should have explained it better in the description. Older versions of TypeScript always end with the same output message: `Compilation complete. Watching for file changes.`, no matter if there were errors or not during the compilation. Newer versions of TypeScript state in the output message if there were errors or not (`Found X errors`). So, to make it work with the older versions, I need to detect whether an error has occurred between the start output message and the complete output message. Otherwise, it will always trigger `onSuccess`, which I think is not a correct behavior. That is why I need `newCompilation` flag above.

**Comment by gilamran** [comment]
Ok, got you. I now starting to remember that :-)
I will do some local tests and approve this PR.

Thanks!!

**Comment by boyanio** [comment]
I have an idea how to enable tests using multiple versions of the tsc, so I can do this in a separate PR later :)

**Comment by gilamran** [comment]
That would be amazing!
